core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java

```diff
@@ -199,12 +201,28 @@ public void offer(RewriteFileGroup group) {
       commitReadyCommitGroups();
     }
 
+    /**
+     * Places a failed file group in the queue to be asynchronously added to {@link
+     * #committerService} when the service has been closed.
+     *
+     * @param group file group had failed
+     */
+    public void failedRewrite(RewriteFileGroup group) {
+      LOG.debug("Offered to failed service: {}", group);
+      Preconditions.checkState(
+          running.get(), "Cannot add rewrites to a service which has already been closed");
+      failedRewrites.add(group);
+    }
+
     /** Returns all File groups which have been committed */
     public List<RewriteFileGroup> results() {
+      List<RewriteFileGroup> results = Lists.newArrayList();
       Preconditions.checkState(
           committerService.isShutdown(),
           "Cannot get results from a service which has not been closed");
-      return committedRewrites;
+      results.addAll(committedRewrites);
+      results.addAll(failedRewrites);
```

Comment:
Not sure we should mix these in the same list? Maybe  we should split this into just two separate lists (or methods)?

Results() ++
Failures() ?

Reply:
Thanks RussellSpitzer,
It has been modified here.